File: solutions/c/kindergarten-garden/1/kindergarten_garden.c
#include "kindergarten_garden.h"
#include <string.h>

plants_t plants(const char* diagram, const char* student) {
    plants_t planted;
    const char* students[12] = {"Alice","Bob","Charlie","David","Eve","Fred","Ginny","Harriet","Ileana","Joseph","Kincaid","Larry"};
    int remains;
    for(int i=0; i < 12; i++) {
        if(strcmp(students[i],student)==0) {
            remains = i;
            break;
        }
    }
    int cups[2] = {remains*2,remains*2+1};
    int j = 0;
    int k = 0;
    int counter = 0;
    while(diagram[k]!='\0' && counter < 4) {
        if(diagram[k] == '\n') {
            j = 0;
            k++;
            continue;
        } else if(j == cups[0] || j == cups[1]) {
            char given = diagram[k];
            if(given == 'C') {
                planted.plants[counter] = CLOVER;
            } else if(given == 'G') {
                planted.plants[counter] = GRASS;
            } else if(given == 'R') {
                planted.plants[counter] = RADISHES;
            } else if(given == 'V') {
                planted.plants[counter] = VIOLETS;
            }
            counter++;
        }
        j++;
        k++;
    }
    return planted;
}

File: solutions/julia/kindergarten-garden/1/kindergarten-garden.jl
function plants(diagram, student)
    class = ["Alice","Bob","Charlie","David","Eve","Fred","Ginny","Harriet","Ileana","Joseph","Kincaid","Larry"]
    new = split(diagram,'\n')
    top = new[1]
    bottom = new[2]
    base = 1
    for classmate in class
        if classmate == student
            front = base*2-1
            list = [top[front],top[front+1],bottom[front],bottom[front+1]]
            return encoding.(list)
        else
            base += 1
        end
    end
    if base > length(class) throw(ArgumentError("Not a valid student.")) end
end

function encoding(letter)
    code = ['G','C','R','V']
    plants = ["grass","clover","radishes","violets"]
    comprehension = Dict(zip(code,plants))
    return comprehension[letter]
end